### ISSUE TITLE
`window.m` is set to "4" after the first tick

### ISSUE THREAD

**Issue by pygy** [jsbin/jsbin#2789]
http://jsbin.com/nejemasego/edit?js,console

```JS
m = 5;
console.log("sync", m);

setTimeout(function () {
  console.log("async", m);
}, 0);
```

This will log `5` then `"4"` when the code is automatically run (either on load or after editing). If you click the "run" button in the "console" pane, you get `5` the in async code as well.

It makes it very annoying to use the Mithril framework with JSBin, since it uses `m` as its namespace.

**Comment by pygy** [comment]
Also, FWIW, [not everyone is affected by this, even though I'm not alone](https://github.com/lhorie/mithril.js/issues/1091#issuecomment-224004225).

It happens in multiple browsers, on a Mac.

**Comment by pygy** [comment]
Digging further, it only happens if a global `m` is set manually:

```JS
var m = 5; // comment this out and nothing is printed by the 'timeout' code.
(function(){
  var keys = Object.keys(window);
  var kv = keys.reduce(function(acc, k){acc[k] = window[k];return acc;},{});
  setTimeout(function(){
    console.log('timeout');
    k2 = Object.keys(window);
    k2.forEach(function(k){
      if (keys.indexOf(k)===-1 || window[k] !== kv[k]) console.log(k, window[k],'----');
    });
  },0);
}());
```

**Comment by remy** [comment]
Add /quiet to the URL. Does it still happen?

**Comment by pygy** [comment]
http://jsbin.com/nejemasego/quiet/edit?js,console also goes `5`, `"4"`, ...

**Comment by pygy** [comment]
Also, rather that `m = 5`, I've tried to use 

```JS
Object.defineProperty(window, 'm', {
  get: function() { return 5; },
  set: function(newValue) { throw new Error("don't set m"); },
  enumerable: true,
  configurable: true
});
```

Which works in the console (`m` returns `5`, setting it throws) but gives 

```
TypeError: can't redefine non-configurable property "m"
<anonymous>   jijatorapa.js:15
k</b.render/<()   runner-3.35.13.min.js:1
j</a.use/<()   runner-3.35.13.min.js:1
```

When put in the JS Bin editor.

**Comment by remy** [comment]
<img width="938" alt="screen shot 2016-07-21 at 19 56 08" src="https://cloud.githubusercontent.com/assets/13700/17035027/3d2cfcd0-4f7d-11e6-8bd3-a32caba34875.png">

Tried your bin out and it _looks_ fine to me. I'm going to close as it might be an edge case and I'm pretty sure it's not related to jsbin. 

That said, happy to continue to discuss if you think this is wrong.

**Comment by tivac** [comment]
I'm still seeing the 5, 4, 4, 4 behavior in Brave, Chrome Canary, and Firefox Dev edition (weird choices I know, but what I've got handy).

I don't think this is fixed.

**Comment by tivac** [comment]
If I add a `debugger` statement right after the first `console.log()`, then start stepping through I end up in some VM code.

(This has been formatted by the editor, I'm not sure where it actually came from)

```js
for (var m in __blocked.methods) {
    try {
        window[m] = __blocked.old[m];
        delete __blocked;
    } catch (e) {}
}
;
```

which rewrites `window.m` to "0", "1", "2", "3", "4" in order. Looks like the global method safety code is tripping over the assignment for some reason?

**Comment by pygy** [comment]
Yup, it's still there in Chrome, Safari and Firefox too (when loading or after editing the code the page, not when pressing "run".

Of note, the auto-run/live-reload/hot-shebang thing doesn't work in Chrome over HTTP, only over HTTPS. https://jsbin.com/nejemasego/edit?js,console

**Comment by pygy** [comment]
It also occurs in IE 11 and Edge on a different machine. This is definitely not a fluke.

**Comment by remy** [comment]
Okay, so I think I know how to replicate this, and I think I understand why it's happening.

It happens if you check the "auto-run js" in the output panel. It doesn't run when you do cmd+enter or "run with js". 

The problem is: the blocking prompt nuking code is leaking a variable, and it's getting caught in your code.

I'll fix that up.

**Comment by remy** [comment]
I believe with this fix, in v3.38.10, you should find it's fixed.

Thanks so much for hanging in there.

**Comment by pygy** [comment]
Hooray, many thanks to you (and @tivac for pointing out the source of the issue).
